tryoutmanager/models.py

```diff
@@ -0,0 +1,139 @@
+                                  UUID,
+                                  ForeignKey('challenges.id')))
+
+challenges_submissions = Table('challenges_submissions',
+                               Base.metadata,
+                               Column('challenge_id',
+                                      UUID,
+                                      ForeignKey('challenges.id')),
+                               Column('submission_id',
+                                      UUID,
+                                      ForeignKey('submissions.id')))
+
+
+# Tables corresponding to concrete models
+class AdminUser(Base):
+    __tablename__ = 'admin_users'
+
+    email = Column(String, nullable=False, unique=True, primary_key=True)
+    password = Column(String, nullable=False)
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+
+
+class Marker(Base):
+    __tablename__ = 'markers'
+
+    email = Column(String, nullable=False, unique=True, primary_key=True)
+    password = Column(String, nullable=False)
+    name = Column(String, nullable=False)
+    challenges = relationship('Challenges',
+                              secondary=markers_challenges,
+                              back_populates='markers')
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+
+
+class User(Base):
+    __tablename__ = 'users'
+
+    email = Column(String, primary_key=True)
+    password = Column(String, nullable=False)
+    name = Column(String, nullable=False)
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+    confirmed = Column(Boolean, nullable=False, default=False)
+    email_confirmation_slug = Column(String, unique=True)
+    email_confirmation_expiry = Column(DateTime(timezone=True))
+
+
+class Challenge(Base):
+    __tablename__ = 'challenges'
+
+    id = Column(UUID, default=uuid.uuid4, primary_key=True)
+    name = Column(String, nullable=False, unique=True)
+    markers = relationship('Markers',
+                           secondary=markers_challenges,
+                           back_populates='challenges')
+    submissions = relationship('Submission',
+                               secondary=challenges_submissions,
+                               back_populates='submissions')
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+
+
+class Submission(Base):
+    __tablename__ = 'submissions'
+
+    id = Column(UUID, default=uuid.uuid4, primary_key=True)
+    contestant = relationship('User', back_populates='submissions')
+    contestant_email = Column(String,
+                              ForeignKey('users.email'),
+                              nullable=False)
+    challenge = relationship('Challenge', back_populates='submissions')
+    challenge_id = Column(UUID,
+                          ForeignKey('challenges.id'),
+                          nullable=False)
+    reservation = relationship('Reservation')
+    status = Column(Enum('open',
```

Comment:
A submission can be:
- Submitted
- Late submit
- In Progress

There might be others I'm missing

Reply:
I thought that would get inferred from the git history, but a helper column would be nice to have. Should this be distinguished from a window being open or closed? (In that case, it makes more sense to move _this_ `status` column over to `Reservation`).

Reply:
See other comment.